### ISSUE TITLE
Can't be built with latest version of nginx (tested with 1.12.1)

### ISSUE THREAD

**Issue by dataviruset** [sflow/nginx-sflow-module#8]
Tested with nginx 1.12.1, but I guess it has been broken for a while.

```
	-o objs/addon/nginx-sflow-module-0.9.10/ngx_http_sflow_config.o \
	../nginx-sflow-module-0.9.10/ngx_http_sflow_config.c
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c:78:5: error: unknown type name 'ngx_mutex_t'
     ngx_mutex_t *mut;
     ^
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c: In function 'ngx_http_sflow_tick_handler':
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c:282:9: error: implicit declaration of function 'ngx_mutex_lock' [-Werror=implicit-function-declaration]
         SFWB_LOCK(cf->sfwb);
         ^
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c:282:19: error: 'cf' undeclared (first use in this function)
         SFWB_LOCK(cf->sfwb);
                   ^
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c:79:39: note: in definition of macro 'SFWB_LOCK'
 #define SFWB_LOCK(_s) ngx_mutex_lock((_s)->mut)
                                       ^
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c:282:19: note: each undeclared identifier is reported only once for each function it appears in
         SFWB_LOCK(cf->sfwb);
                   ^
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c:79:39: note: in definition of macro 'SFWB_LOCK'
 #define SFWB_LOCK(_s) ngx_mutex_lock((_s)->mut)
                                       ^
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c:285:9: error: implicit declaration of function 'ngx_mutex_unlock' [-Werror=implicit-function-declaration]
         SFWB_UNLOCK(cf->sfwb);
         ^
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c: In function 'sfwb_init':
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c:872:5: error: implicit declaration of function 'ngx_mutex_init' [-Werror=implicit-function-declaration]
     sm->mut = ngx_mutex_init(cf->log, 0);
     ^
../nginx-sflow-module-0.9.10/ngx_http_sflow_module.c:872:13: error: assignment makes pointer from integer without a cast [-Werror]
     sm->mut = ngx_mutex_init(cf->log, 0);
             ^
cc1: all warnings being treated as errors
make[1]: *** [objs/addon/nginx-sflow-module-0.9.10/ngx_http_sflow_module.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/root/rpmbuild/BUILD/nginx-1.12.1'
make: *** [build] Error 2
```

**Comment by sflow** [comment]
Thanks. I'll take a look.

**Comment by sflow** [comment]
It looks like src/os/unix/ngx_thread.h had changed in cosmetic ways  (e.g. "ngx_mutex_t" became "ngx_thread_mutex_t").   I checked in some changes to accommodate this and now it compiles OK.   Please can you "git pull" and let me know if this works for you now?   If so,  I'll make a formal release.

As a meta-comment,  it looks as though the Nginx developers are moving away from the idea of supporting worker-threads.

**Comment by MariaDima** [comment]
Hello,

I faced the same issue, while compiling nginx 1.13.6.
I patched the latest release (release-0.9.10) with your changes and compilation succeeded.
It will be good to have a new release including your fix.

Thanks,
Maria

**Comment by sflow** [comment]
I added a new release.
